### PR TITLE
[AUT-117] - Add modal support for enrolment

### PR DESCRIPTION
**Pull request by hwhmeikle**
- Deprecate `challenge()` and `mfa()`
- Add `launch()`
- Remove `authsignal-challenge-success` and `authsignal-challenge-success` event listener
- Add listener for `AuthsignalWindowMessage.AUTHSIGNAL_CLOSE_POPUP` event
- Add `endpoint` option so that we can verify event messages come from a trusted source

### CODE REVIEW

File: src/authsignal-browser.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  }
+
+  /**
+   * @deprecated Use launch() instead.
+   */
+  mfa(challenge: {mode?: "redirect"} & MfaInput): undefined;
+  mfa(challenge: {mode: "popup"} & MfaInput): Promise<boolean>;
+  mfa({mode, url}: MfaInput) {
+    if (mode === "popup") {
```
Comment (hwhmeikle) [inline src/authsignal-browser.ts:44]: Had to do this `if` to make typescript happy due to function overload things

Comment (hwhmeikle) [reply src/authsignal-browser.ts:44]: Could potentially do a `@ts-expect-error` here instead

File: src/authsignal-browser.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      window.location.href = url;
+    } else {
+      const Popup = new PopupHandler();
+
+      Popup.show({url});
+
+      return new Promise<boolean>((resolve) => {
+        const onMessage = (event: MessageEvent) => {
+          if (event.origin === this.endpoint) {
+            if (event.data === AuthsignalWindowMessage.AUTHSIGNAL_CLOSE_POPUP) {
```
Comment (hwhmeikle) [inline src/authsignal-browser.ts:77]: Only listening for the single message now, and never rejecting the promise

**Review by chrisfisher — APPROVED** [review]
Nice

**Pull request merged**